### PR TITLE
 Localize error messages

### PR DESCRIPTION
**Pull request by palagdan**
Resolves partially kbss-cvut/fta-fmea-ui#550

### CODE REVIEW

File: src/main/java/cz/cvut/kbss/analysis/service/UserRepositoryService.java [hunk 1]
```diff
@@ -63,12 +63,12 @@ public void updateCurrent(UserUpdateDTO userUpdate) {
         User currentUser = getCurrentUser();
         if (!currentUser.getUri().equals(userUpdate.getUri())) {
             log.warn("< updateCurrent - URIs do not match! {} != {}", currentUser.getUri(), userUpdate.getUri());
-            throw new LogicViolationException("User update uri does not match current user!");
+            throw new LogicViolationException("errors.user.update.uriMismatch","User update uri does not match current user!");
```
Comment (blcham) [inline src/main/java/cz/cvut/kbss/analysis/service/UserRepositoryService.java:66]:
```suggestion
            throw new LogicViolationException("error.user.update.uriMismatch","User update uri does not match current user!");
```

Comment (blcham) [reply src/main/java/cz/cvut/kbss/analysis/service/UserRepositoryService.java:66]: Everywhere else you have `error` but here `errors`

**Review by blcham — APPROVED** [review]
LGTM, @kostobog have a look as well pls

File: src/main/java/cz/cvut/kbss/analysis/service/UserRepositoryService.java (hunk 1 above)
Comment (palagdan) [reply src/main/java/cz/cvut/kbss/analysis/service/UserRepositoryService.java]: done!

**Comment by blcham** [comment]
@kostobog hope this was ok to merge

**Comment by kostobog** [comment]
@blcham 
I guess it is ok. We should consider resolving the issues mentioned in the comments above.

**Comment by palagdan** [comment]
@blcham @kostobog 
I didn't provide messageIds for following exceptions:

- (CalculationException, EntityNotFoundException, FtaFmeaException) because the message is generated on the backend and are not just plain text.

- (ExternalServiceException, PersistenceException, ResourceException) - These are 500 status exceptions, so it shouldn't provide any message explanation to the user, only logs for developer. Please correct me if I am wrong.

**Pull request merged**